File: 2015/day5/day5.js
const fs = require("fs");
const input = fs.readFileSync("input").toString('utf-8').split('\n')

const day5 = (input, func) => input.filter(func).length

const niceString = line => {
  let vowels = line.match(/[aieou]/g)
  let disallowed = line.match(/ab|cd|pq|xy/)
  let doubles    = line.match(/([a-z])\1/)

  return (vowels !== null && vowels.length > 2) && 
         (disallowed === null || disallowed.length === 0) &&
         (doubles !== null && doubles.length > 0)
}

const newNiceString = line => {
  return false
}

console.log("Part 1:", day5(input, niceString))
console.log("Part 2:", day5(input, newNiceString))

File: 2015/day5/day5.py
import re

with open("input", "r+") as file:
  f = file.read().split("\n")

sum = 0
for line in f:
  vowels = re.findall(r'[aeiou]', line)
  disallowed = re.search(r'ab|cd|pq|xy', line)
  doubles = re.search(r'(.)\1', line)

  if (len(vowels) > 2) and (disallowed is None) and (doubles is not None):
    sum += 1

print("Part 1:", sum)

sum = 0
for line in f:
  repeat = re.search(r'(..).*\1', line)
  sandwich = re.search(r'(.).\1', line)

  if (repeat is not None) and (sandwich is not None):
    sum += 1

print("Part 2:", sum)